File: frontend/src/types/salesChannel.ts
export interface SalesChannel {
    id: number;
    code: string;
    name: string;
    platform_type: string; // marketplace, social, branch, other
    is_active: boolean;
    description?: string;
    config?: string;
    created_at: string;
    created_by_name?: string;
    updated_at: string;
}

export interface CreateSalesChannelInput {
    code: string;
    name: string;
    platform_type: string;
    description?: string;
}

export interface UpdateSalesChannelInput {
    name?: string;
    platform_type?: string;
    is_active?: boolean;
    description?: string;
}

export interface SalesChannelFilters {
    platform_type?: string;
    is_active?: boolean;
    search?: string;
    offset?: number;
    limit?: number;
}

export interface SalesChannelListResponse {
    data: SalesChannel[];
    total: number;
}

File: backend/internal/dto/sales_channel.go
package dto

// CreateSalesChannelRequest is the DTO for creating a new sales channel
type CreateSalesChannelRequest struct {
	Code         string `json:"code" binding:"required"`
	Name         string `json:"name" binding:"required"`
	PlatformType string `json:"platform_type" binding:"required"` // marketplace, social, branch, other
	Description  string `json:"description"`
}

// UpdateSalesChannelRequest is the DTO for updating a sales channel
type UpdateSalesChannelRequest struct {
	Name         string `json:"name"`
	PlatformType string `json:"platform_type"`
	IsActive     *bool  `json:"is_active"`
	Description  string `json:"description"`
}

// SalesChannelFilterRequest is the DTO for filtering sales channels
type SalesChannelFilterRequest struct {
	PlatformType string `form:"platform_type"`
	IsActive     *bool  `form:"is_active"`
	Search       string `form:"search"`
	Offset       int    `form:"offset,default=0"`
	Limit        int    `form:"limit,default=20"`
}
